lib/parse.js

```diff
@@ -30,16 +37,27 @@ var parseValues = function parseQueryStringValues(str, options) {
 
         var key, val;
         if (pos === -1) {
-            key = options.decoder(part, defaults.decoder);
+            key = options.decoder(part, defaults.decoder, charset);
             val = options.strictNullHandling ? null : '';
         } else {
-            key = options.decoder(part.slice(0, pos), defaults.decoder);
-            val = options.decoder(part.slice(pos + 1), defaults.decoder);
+            key = options.decoder(part.slice(0, pos), defaults.decoder, charset);
+            val = options.decoder(part.slice(pos + 1), defaults.decoder, charset);
         }
-        if (has.call(obj, key)) {
-            obj[key] = [].concat(obj[key]).concat(val);
+        if (key === 'utf8' && options.utf8Sentinel) {
```

Comment:
so the `utf-8` key would only matter with the `utf8Sentinel` option?

Reply:
Should the user be able to customize the name of this parameter or is this following some specific standard?

Reply:
It's not an actual standard, but `utf8=✓` seemed to be a de facto standard back when I made this originally. Back when the internet was still in the process of switching to utf-8 and iso-8859-1 was more prevalent :)

* https://softwareengineering.stackexchange.com/questions/168751/is-the-use-of-utf8-preferable-to-utf8-true
* Ruby on rails added it to all forms at some point, I don't know if they still do
* I saw it in Google Search urls, that was how I became aware of the trick. They don't use it anymore, though, not even if I make a search in IE6 on Windows XP.

I don't think it'd be relevant to let the user customize the parameter name.

Reply:
Gotcha. Can you add this to the readme somewhere? For example how will folks know how to use it, especially if it's not an actual standard they may have learned about elsewhere?

Reply:
Another option is to make `true` mean `"utf8"`, but make a string value set *that* to be the sentinel? (we'd still want it in the readme)

Reply:
Will get back to this soon and add docs etc. Still not sure I see the value of customizing the parameter name etc. Feels like a "when in doubt, leave it out" scenario until someone actually requests it.

I just noticed that github is also using `utf8=✓`. If you go to "Files changed" in this PR, then change the diff settings, you'll end up at a url like this:

https://github.com/ljharb/qs/pull/268/files?utf8=%E2%9C%93&diff=unified

... so RoR probably still does it by default.

Reply:
I think that making it maximally configurable right out of the gate, but also pleasant for the common defaults, is a good idea.

Reply:
Added a bit of docs in 452008dc35fb159ea684c6f9ade43d3f5251d68c.

Reply:
I'm still not sure I get why maximum configurability for the `utf8Sentinel` parameter would be a good thing when we don't know that anyone will ever need it. That makes it easy to get the configuration parameters wrong, but at that point there'd be no going back. Also it'd add complexity because we'd need to dynamically compute the misinterpreted-as-iso-8859-1 versions of it.